### PR TITLE
Minjiaz/compression gpt

### PR DESCRIPTION
**Pull request by minjiaz**
This PR contains GPT pretraining distillation and quantization examples. It also contains the recent updated changes related to compressor API.

### CODE REVIEW

File: pretrain_gpt.py [hunk 1]
```diff
@@ -217,10 +233,11 @@ def forward_step(data_iterator, model, teacher_model=None):
     moe_loss = sum(moe_losses) * args.moe_loss_coeff
 
     mos_loss = 0
-    if args.mos:
+    if args.mos or args.kd:
         assert model.training
         mos_loss = calculate_mos_loss(args, stu_output, teacher_model, tokens, position_ids, attention_mask)
     
+    print_rank_0('*** args.kd {}, kd loss {}'.format(args.kd, mos_loss))
```
Comment (conglongli) [inline pretrain_gpt.py:240]: This seems to be a debug logging and no longer needed? It will print under non-kd cases, and there is a "print_rank_0('>>> total loss: {}, lm loss {}, kd loss {}'.format(loss, averaged_loss[0], mos_loss))" above in the same file.

**Review by conglongli — APPROVED** [review]
Thanks @minjiaz for adding the changes we discussed. I left some minor comments.
